src/fabric/test/fabric2_doc_crud_tests.erl

```diff
@@ -765,6 +769,101 @@ create_local_doc_random_rev({Db, _}) ->
     ?assertEqual(Doc5#doc{revs = {0, [<<"2">>]}}, Doc6).
 
 
+create_a_large_local_doc({Db, _}) ->
+    UUID = fabric2_util:uuid(),
+    LDocId = <<?LOCAL_DOC_PREFIX, UUID/binary>>,
+    Body = << <<"x">> || _ <- lists:seq(1, 300000) >>,
+    Doc1 = #doc{
+        id = LDocId,
+        revs = {0, []},
+        body = Body
+    },
+    ?assertEqual({ok, {0, <<"1">>}}, fabric2_db:update_doc(Db, Doc1)),
+    {ok, Doc2} = fabric2_db:open_doc(Db, Doc1#doc.id, []),
+    ?assertEqual(Doc1#doc{revs = {0, [<<"1">>]}}, Doc2),
+
+    % Read via fold_local_docs
+    {ok, Result} = fabric2_db:fold_local_docs(Db, fun(Data, Acc) ->
+        case Data of
+            {row, [{id, DocId} | _]} when LDocId =:= DocId ->
```

Comment:
Should we assert that rev is correct in these fold loops?

Reply:
Good idea. Asserted the rev in the `?assertEqual` statements below